app/models/package_manager/go.rb

```diff
@@ -87,17 +87,21 @@ def self.one_version(raw_project, version_string)
     end
 
     def self.project(name)
+      # try and find the canonical name from the Go proxy and use that if we find it
```

Comment:
might be worth mentioning that the reasoning is because pkg.go.dev has been known to store the wrong casings of a Go Module name, eg https://pkg.go.dev/github.com/go-sql-driver/mysQL